src/System.d.ts

```diff
@@ -4,19 +4,34 @@ import {Component, ComponentConstructor} from "./Component";
  * A system that manipulates entities in the world.
  */
 export abstract class System {
+  /**
+   * Defines what Components the System will query for.
+   * This needs to be user defined.
+  */
+  static queries: {
+    [queryName: string]: Component[],
```

Comment:
After I read class System. I think queries might be: 

```
static queries: {
  [queryName: string]: { 
    components: Component[],
    mandatory?: boolean, 
    listen?: "added" | "removed" | "changed"
  }
}
```

For field `components`, I see in example that this field accepted component query but the component query in [this](https://github.com/MozillaReality/ecsy/blob/dev/src/System.d.ts#L29) file, the `Not` selector returns `object` not components. I'm not sure what should we do in this case.

Reply:
Hrmm, is `listen` also an array of those 3 string consts?

Reply:
@dannyfritz I see System for each valid events that declared in this [line](https://github.com/MozillaReality/ecsy/blob/dev/src/System.js#L55) and get them from `listen`. 

Please suggest me if I'm wrong.

Reply:
Ah yeah, it appears to be:
```js
listen?: {
  added?: boolean,
  removed?: boolean,
  changed?: boolean,
}
```

Reply:
Ah you're correct. :)

Reply:
This isn't valid TypeScript. :cry: 
```ts
  static queries: {
    [queryName: string]: (Component | NotComponent)[],
    listen?: {
      added?: Boolean,
      removed?: Boolean,
      changed?: Boolean | Component[],
    },
  };
```

Reply:
Also not allowed :frowning_face: 
```ts
static queries: {
    [queryName: Exclude<string, "listen">]: (Component | NotComponent)[],
    listen?: {
      added?: Boolean,
      removed?: Boolean,
      changed?: Boolean | Component[],
    },
  };
```

Reply:
Oh, this typing is just wrong. XD. Will fix.

Reply:
got it:

```ts
  static queries: {
    [queryName: string]: {
      components: (Component | NotComponent)[],
      listen?: {
        added?: Boolean,
        removed?: Boolean,
        changed?: Boolean | Component[],
      },
    }
  };
```